Jenkinsfile

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env groovy
+@Library('github.com/zanata/zanata-pipeline-library@master')
+def dummyForLibrary
+
+/**
+ * Jenkinsfile for zanata-platform
+ */
+
+// It seems that we need a node to access env.DEFAULT_NODE
+node {
+  def projectProperties = [
+    [
+      $class: 'BuildDiscarderProperty',
+      strategy: [$class: 'LogRotator',
+        daysToKeepStr: '30',       // keep records no more than X days
+        numToKeepStr: '20',        // keep records for at most X builds
+        artifactDaysToKeepStr: '', // keep artifacts no more than X days
+        artifactNumToKeepStr: '4'] // keep artifacts for at most X builds
+    ],
+    [
+      $class: 'ParametersDefinitionProperty',
+      parameterDefinitions: [
+        [
+          $class: 'LabelParameterDefinition',
+          /* Specify the default node in Jenkins environment DEFAULT_NODE
+           * or default to build on any node
+           */
+          defaultValue: env.DEFAULT_NODE ?: 'master || !master',
+          description: 'Node label that allow to build',
+          name: 'LABEL'
+        ]
+      ]
+    ],
+  ]
+
+  properties(projectProperties)
+}
+
+def surefireTestReports='target/surefire-reports/TEST-*.xml'
+
+/* Upto stash stage should fail fast:
+ * Failed and stop the build
+ * Yet able to create report
+ */
+timestamps {
+  node(LABEL) {
+    ansicolor {
+      try {
+        stage('Checkout') {
+          info.printNode()
+          notify.started()
+
+          // Shallow Clone does not work with RHEL7, which use git-1.8.3
+          // https://issues.jenkins-ci.org/browse/JENKINS-37229
+          checkout scm
+          // Clean the workspace
+          sh "git clean -fdx"
+        }
+
+        // Build and Unit Tests
```

Comment:
We would, but Maven isn't really designed to run lifecycles phases in isolation, because it always wants to run the earlier lifecycle phases (eg `mvn test` includes the work done by `mvn compile`).

Besides, profiling is better handled with Maven features than with Jenkins pipelines stages.